source/websocket.c

```diff
@@ -1373,16 +1373,14 @@ static int s_decoder_on_payload(struct aws_byte_cursor data, void *user_data) {
 
 /* Invoke user cb */
 static int s_decoder_on_user_payload(struct aws_websocket *websocket, struct aws_byte_cursor data) {
-    if (!websocket->on_incoming_frame_payload) {
-        return AWS_OP_SUCCESS;
-    }
+    if (websocket->on_incoming_frame_payload) {
+        if (!websocket->on_incoming_frame_payload(
```

Comment:
really minor: does it have to be 2 ifs? if (cb && !cb(...)) seems like a common pattern

Reply:
looks like the old code was exiting early if no callback was defined and now we are updating window and doing other things. was it a bug in old code?

Reply:
All the callbacks in this file are invoked with 2 ifs: one to check whether to invoke the callback, and the next to invoke it and see if it fails

I think it's easier to read